Enable bomb crash screen on Cervantes

tjaderxyz opened the pull request:
I have made the minimal modifications to enable the bomb crash screen on Cervantes while preserving as much of the current behavior as possible, but the current behavior is inconsistent: it leaves the restart loop for return values between 1 and 84 but treats any return value greater than 87 as a request to restart koreader.

I decided to maintain the spirit of the original script and only restart koreader for the special return values (85, 86 and 87), and exit for any other return value, showing the crash screen if the return value is not zero.

If it is deemed better that koreader.sh loops forever restarting koreader until the return value is zero I can change the code.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/koreader/7731)
<!-- Reviewable:end -->

pazos commented:
>  but the current behavior is inconsistent: it leaves the restart loop for return values between 1 and 84 but treats any return value greater than 87 as a request to restart koreader.

Yup, my bad. The old code was checking the special snowflake `85` used to signal a restart in koreader scripts. I considered everything above 85 as snowflakes and defined a couple for the platform. Every other non-return code will exit KO because that's expected on both launch modes (standalone/ssh from QBookApp).

Now, the crash bomb is also a snowflake to handle within the restart loop.

> I decided to maintain the spirit of the original script and only restart koreader for the special return values (85, 86 and 87), and exit for any other return value, showing the crash screen if the return value is not zero.

That's a great contribution.

> If it is deemed better that koreader.sh loops forever restarting koreader until the return value is zero I can change the code.

The Kobo implementation looks quite ok. Keep a crash count and break after a few errors. Break would mean return to QBookApp or executing `koreader-standalone's main loop` again, depending on the launch method.

pazos reviewed (changes requested):
Please use spaces instead of tabs

tjaderxyz commented:
> Please use spaces instead of tabs

I didn't find any tabs in my commits. What I did notice is that I used 8 spaces instead of four. I copied it from kobo's where it was an if inside a while and forgot to remove one level of indentation, I will fix that soon.

pazos commented:
@tjaderxyz

I'm attaching here ci checks, in case you can't see the test report.

```
Running shellcheck on platform/cervantes/koreader.sh

In platform/cervantes/koreader.sh line 95:
    "${RETURN_VALUE}" -eq "${ENTER_USBMS}" -o \
                                           ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.


In platform/cervantes/koreader.sh line 96:
    "${RETURN_VALUE}" -eq "${ENTER_QBOOKAPP}" -o \
                                              ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.

For more information:
  https://www.shellcheck.net/wiki/SC2166 -- Prefer [ p ] || [ q ] as [ p -o q...
Running shfmt on platform/cervantes/koreader.sh
Warning: platform/cervantes/koreader.sh does not abide by coding style, diff for expected style:
--- platform/cervantes/koreader.sh.orig
+++ platform/cervantes/koreader.sh
@@ -156,7 +156,7 @@
     ./fbink -q -f -s
     # Cue a lemming's faceplant sound effect!
 
-    timeout 15 head -c 24 /dev/input/event1 > /dev/null
+    timeout 15 head -c 24 /dev/input/event1 >/dev/null
 fi
 
 if [ "${STANDALONE}" != "true" ]; then
```

pazos reviewed (changes requested):
Please keep the logic of kobos: https://github.com/koreader/koreader/blob/master/platform/kobo/koreader.sh#L340-L420

Also keep the same message as in other platforms.

tjaderxyz commented:
Sure, will do it tomorrow. I removed the crash count and changed the
message to conform better to the current behavior before asking for
comments.

Martín Fernández ***@***.***> schrieb am Sa., 22. Mai 2021,
22:46:

> ***@***.**** requested changes on this pull request.
>
> Please keep the logic of kobos:
> https://github.com/koreader/koreader/blob/master/platform/kobo/koreader.sh#L340-L420
>
> Also keep the same message as in other platforms.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/koreader/koreader/pull/7731#pullrequestreview-666221942>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAT74YTBDAUFQMPAMW3YXLTPBM7TANCNFSM45J4Z3TA>
> .
>

tjaderxyz commented on platform/cervantes/koreader.sh:
```diff
@@ -91,20 +94,101 @@ ENTER_QBOOKAPP=87
+        {
+            echo "!!!!"
+            echo "Uh oh, something went awry... (Crash n°${CRASH_COUNT}: $(date +'%x @ %X'))"
+            echo "Running FW $(xargs </.version) on Linux $(uname -r) ($(uname -v))"
```
Does anyone know a better way to get the FW version on Cervantes? On my cervantes 4 this returns "246 71d478", which may or may not be useful.

Frenzie replied:
I don't know anything about the platform as such but 246 looks like it should be simple and meaningful?

pazos replied:
There's no way to retrieve the FW version. You can:

1. print the HWConfig version: `echo "Running FW $(ntxinfo /dev/mmcblk0 | grep version_name | cut -d ":" -f2)  on Linux $(uname -r) ($(uname -v))"`

2. print the debian version: `echo "Running Debian $(cat /etc/debian_version). Linux $(uname -r) ($(uname -v))"`

tjaderxyz replied:
And which of these three options are actually useful to show? We could show any combination of them or not show anything at all.

NiLuJe replied:
FWIW, this made sense on Kobo, it might not here, so I wouldn't necessarily be overly attached to it. I'll let @pazos be the judge of that ;).

pazos replied:
> And which of these three options are actually useful to show?

The company is dead and each supported device has **one** FW version. So

> not show anything at all.

pazos reviewed (approved):
lgtm

Frenzie commented on platform/cervantes/koreader.sh:
```diff
@@ -125,8 +211,10 @@ while [ "${RETURN_VALUE}" -ge "${RESTART_KOREADER}" ]; do
             sleep 10
         done
     fi
+
```
```suggestion
```

Frenzie commented on platform/cervantes/koreader.sh:
```diff
@@ -125,8 +211,10 @@ while [ "${RETURN_VALUE}" -ge "${RESTART_KOREADER}" ]; do
+
 done
 
+
```
```suggestion
```
